app/models/shipit/deploy.rb

```diff
@@ -136,6 +141,27 @@ def confirmed?
       confirmations.abs >= CONFIRMATIONS_REQUIRED
     end
 
+    def too_dangerous?
+      revert_being_left_behind?
+    end
+
+    def revert_being_left_behind?
+      return if commits_since.empty?
+
+      commits_since.each do |commit|
+        matches = commit.message.match(/^\s*Revert "(.*)"\s*$/)
+        next unless matches
+        parent_message = matches[1]
+
+        commits.each do |other_commit|
+          next if commit == other_commit
+          return true if parent_message == other_commit.message
```

Comment:
Hmmm now that I look at this again, I think this needs to be a bit smarter and ignore the "Merge pull request ..." prefix that the parent might have. Will update this PR on Monday.